docs/zkapps/tutorials/10-account-updates.mdx

```diff
@@ -0,0 +1,381 @@
+
+### Smart Contracts
+
+We'll be buliding two smart contracts to use in our example:
+
+First, [ProofsOnlyZkApp.ts](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/10-account-updates/src/ProofsOnlyZkApp.ts).
+
+This is a zkApp configured to only be modifiable or modifiable via proofs (and for the zkApp we're deploying, is not upgradable once deployed). This means that while someone must own the private key to deploy the zkApp itself - after deployed, only the rules in the zkApp determine how the zkApp functions. The private key is no longer useful for anything.
+
+This zkApp also has functions that call other functions, so we can explore how that impacts a transcation's AccountUpdates.
+
+We'll start by adding the main contents of the zkApp:
+
+```typescript
+
+export class ProofsOnlyZkApp extends SmartContract {
+  @state(Field) num = State<Field>();
+  @state(Field) calls = State<Field>();
+
+  deploy(args: DeployArgs) {
+    super.deploy(args);
+    this.setPermissions({
+      ...Permissions.default(),
+      setDelegate: Permissions.proof(),
+      setPermissions: Permissions.proof(),
+      setVerificationKey: Permissions.proof(),
+      setZkappUri: Permissions.proof(),
+      setTokenSymbol: Permissions.proof(),
+      incrementNonce: Permissions.proof(),
+      setVotingFor: Permissions.proof(),
+      setTiming: Permissions.proof(),
+    });
+  }
+
+  @method init() {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertFalse();
+
+    super.init();
+    this.num.set(Field(1));
+    this.calls.set(Field(0));
+  }
+
+  ...
+```
+
+This configures the zkApp as described, and initializes the zkApp with the values we'd like. 
+
+:::caution
+Note we assert that `provedState` is `False` in `init`. This ensures that `init` cannot be called again after the zkApp is setup. Without this, our zkApp could be reset by anyone calling the `init` transaction on our zkApp, and is recommended for most zkApps.
+:::
+
+Next, we'll add two functions:
+
+```typescript
+  ...
+
+  @method add(incrementBy: Field) {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const num = this.num.get();
+    this.num.assertEquals(num);
+    this.num.set(num.add(incrementBy));
+
+    this.incrementCalls();
+  }
+
+  @method incrementCalls() {
+    this.account.provedState.assertEquals(this.account.provedState.get());
+    this.account.provedState.get().assertTrue();
+
+    const calls = this.calls.get();
+    this.calls.assertEquals(calls);
+    this.calls.set(calls.add(Field(1)));
+  }
+
+  ...
+```
+
+:::caution
+In each of these, we are also asserting `provedState` is `true`. This ensures the zkApp was initialized as expected, as `provedState` becomes true after `init` is called, and is recommended for most zkApps.
```

Comment:
```suggestion
In each of these methods, you are also asserting `provedState` is `true` to ensure the zkApp was initialized as expected because `provedState` becomes true after `init` is called and is recommended for most zkApps.
```